Src/EB/AMReX_EB_utils.cpp

```diff
@@ -196,11 +196,25 @@ facets_nearest_pt (IntVect const& ind_pt, IntVect const& ind_loop, RealVect cons
 }
 }
 
-void FillSignedDistance (MultiFab& mf, EB2::Level const& ls_lev,
-                         EBFArrayBoxFactory const& eb_factory, int refratio,
+void FillSignedDistance (MultiFab& mf_in, EB2::Level const& ls_lev,
+                         EBFArrayBoxFactory const& eb_factory_in, int refratio,
                          bool fluid_has_positive_sign)
 {
-    AMREX_ALWAYS_ASSERT(mf.is_nodal());
+    AMREX_ALWAYS_ASSERT(mf_in.is_nodal());
+
+    // because the algorithm below is N^2 in the number of points per box,
+    // we always do this operation with a max grid size of 32.
+    auto new_ba = mf_in.boxArray();
+    new_ba.convert(IntVect::TheZeroVector());
```

Comment:
There is no need to convert the index type because BoxArray::maxSize() works for nodal boxes too.